### PR TITLE
[fix] isin : non-contiguous input on cpu

### LINKED ISSUE #67432: isin is incorrect when given non-contiguous inputs on CPU
Run test_noncontiguous_samples_isin_cpu (requires removing the xfail from the OpInfo).

cc @ezyang @gchanan @zou3519 @bdhirsh @jbschlosser @VitalyFedyunin

### PR DESCRIPTION
**Pull request by kshitij12345**
Fixes https://github.com/pytorch/pytorch/issues/67432

### CODE REVIEW

File: aten/src/ATen/native/cpu/TensorCompareKernel.cpp [hunk 1]
```diff
@@ -299,8 +299,10 @@ static void isin_default_kernel_cpu(
   // Since test elements is not an input of the TensorIterator, type promotion
   // must be done manually.
   ScalarType common_type = at::result_type(elements, test_elements);
-  Tensor test_elements_flat = test_elements.to(common_type).ravel();
```
Comment (kshitij12345) [inline aten/src/ATen/native/cpu/TensorCompareKernel.cpp:302]: `ravel` does not necessarily return a contiguous tensor. See #https://github.com/pytorch/pytorch/issues/70657

In anycase, we don't need a contiguous tensor.

**Comment by kshitij12345** [comment]
I think we should also add more sample to check for the `else` branch below

https://github.com/pytorch/pytorch/blob/c468e35d834a1d233a2905cde93dc8f3b6f027dd/aten/src/ATen/native/TensorCompare.cpp#L672-L678

Sample Inputs
https://github.com/pytorch/pytorch/blob/c468e35d834a1d233a2905cde93dc8f3b6f027dd/torch/testing/_internal/common_methods_invocations.py#L6134-L6140

**Comment by mruberry** [comment]
> I think we should also add more sample to check for the `else` branch below
> 
> https://github.com/pytorch/pytorch/blob/c468e35d834a1d233a2905cde93dc8f3b6f027dd/aten/src/ATen/native/TensorCompare.cpp#L672-L678
> 
> Sample Inputs
> 
> https://github.com/pytorch/pytorch/blob/c468e35d834a1d233a2905cde93dc8f3b6f027dd/torch/testing/_internal/common_methods_invocations.py#L6134-L6140

That sounds good, would you like to add a sample input for that case and then ping me to merge this?

**Review by mruberry — APPROVED** [review]
Cool! Thanks @kshitij12345!

**Comment by kshitij12345** [comment]
@mruberry have updated the samples. PTAL :)

**Comment by kshitij12345** [comment]
@mruberry Gentle ping :)

**Comment by mruberry** [comment]
Thanks for the pings, @kshitij12345!

**Pull request closed without merging**